packages/flutter_tools/lib/src/protocol_discovery.dart

```diff
@@ -60,10 +59,13 @@ class ProtocolDiscovery {
 
   void _handleLine(String line) {
     Uri uri;
-    final int index = line.indexOf(_prefix + 'http://');
-    if (index >= 0) {
+
+    final RegExp _regexp = new RegExp('${RegExp.escape(serviceName)} (http://[^ \n]+)');
+    final Match _match = _regexp.firstMatch(line);
+
+    if (_match != null) {
       try {
-        uri = Uri.parse(line.substring(index + _prefix.length));
+        uri = Uri.parse(_match[0]);
```

Comment:
This should be `match[1]` since you want to extract the group that contains the URL